nameko/messaging.py

```diff
@@ -3,40 +3,47 @@
 '''
 from __future__ import absolute_import
 
-import warnings
+import re
 from functools import partial
 from logging import getLogger
 
-import six
-from amqp.exceptions import ConnectionError
-from eventlet.event import Event
-from kombu import Connection
 from kombu.common import maybe_declare
-from kombu.mixins import ConsumerMixin
 
+from nameko import serialization
+from nameko.amqp.consume import Consumer as ConsumerCore
 from nameko.amqp.publish import Publisher as PublisherCore
 from nameko.amqp.publish import get_connection
 from nameko.amqp.utils import verify_amqp_uri
 from nameko.constants import (
-    AMQP_URI_CONFIG_KEY, DEFAULT_HEARTBEAT, HEADER_PREFIX,
-    HEARTBEAT_CONFIG_KEY
+    AMQP_URI_CONFIG_KEY, DEFAULT_HEARTBEAT, DEFAULT_PREFETCH_COUNT,
+    HEADER_PREFIX, HEARTBEAT_CONFIG_KEY, PREFETCH_COUNT_CONFIG_KEY
 )
 from nameko.exceptions import ContainerBeingKilled
-from nameko.extensions import (
-    DependencyProvider, Entrypoint, ProviderCollector, SharedExtension
-)
+from nameko.extensions import DependencyProvider, Entrypoint
 
 
 _log = getLogger(__name__)
 
 
+def encode_to_headers(context_data, prefix=HEADER_PREFIX):
+    return {
+        "{}.{}".format(prefix, key): value
+        for key, value in context_data.items()
+        if value is not None
+    }
+
+
+def decode_from_headers(headers, prefix=HEADER_PREFIX):
+    return {
+        re.sub("^{}\.".format(prefix), "", key): value
+        for key, value in headers.items()
+    }
+
+
```

Comment:
Split this into functions so they can be used without inheriting from `HeaderEncoder `